### DISCUSSION TITLE
Konsole profiles?

### DISCUSSION THREAD

**Discussion by toast003** [nix-community/plasma-manager#63] [Ideas]
I'm thinking of writing a module for setting up konsole, however since it's not part of plasma itself I don't know if it's too out of scope or not

Would anyone like to use that?

**Comment by magnouvean** [comment]
Yeah, I'd say this would be great. Konsole is quite connected with plasma so I think it definitely makes sense to have some sort of module in this project. At the moment I configure it using home-manager dotfiles and lower-level interfaces, but a dedicated interface would be even better and certainly very welcome for a lot of people if you find time to write this :) (If you decide to try this I think a konsolerc high-level interface and a profile interface would make sense)

**Comment by Asqiir** [comment]
Related to that: 

I'm currently playing around with configuring kate.
I could prepare an MR with some mid-level options, would that be in scope for project or should it go somewhere else?

Pro:
* kate is integral part of the plasma environment

Con:
* nobody who does not use plasma can use this module to configure kate (equally applies to konsole, I think. But I'm not sure if k-programs config files are saved in the same way across displaymanagers).

**Reply by magnouvean** [reply]
I'd at least be welcome to modules for applications tightly integrated with kde (like kate). As for the con, I actually don't think this is a problem since I don't actually think there is anything in plasma-manager which requires kde plasma to be installed for it to work, so this shouldn't be a problem.

Most people who use plasma-manager are plasma users naturally though, and for non-plasma kate users it probably is a bit much work to add plasma-manager just for kate. Alternatively it could make sense to create a module for `home-manager` as this will have more reach outside of plasma, but the tooling (like kde config file writing) is probably a little more limited and perhaps a bit harder to do. With that said though, as kate is probably mostly used by kde users, I think it would be in scope for this project as well if there are any difficulties/limitations in implementing it there (`home-manager` usually symlink files which a lot of apps, including a lot of kde apps, are not a fan of).

**Reply by toast003** [reply]
Afaik most people that don´t use plasma also don't use any of the KDE apps ( apart from some exceptions like krita or kdenlive ), so that shouldn't be that big of an issue, however if people want that we could *maybe* eventually move the app config parts to another module. That said tho, that's a lot of work for something that no one seems to want.

**Reply by toast003** [reply]
Btw, I do think it would make more sense to place the options for the kde apps (konsole, kate...) into `programs` instead of `programs.plasma`

That way, if we ever want to move them to another module or even upstream them to home-manager, we will have an easier time doing so
